### PR TITLE
no drinking soup with fork & utensils no longer destroy trash item

### PR DESCRIPTION
**Pull request by alsoandanswer**
you can no longer eat soup/stew with a fork.
also refactors some stew and soups.
adds new trash item - empty stew pot.
adds a useful proc to snacks : is_liquid.

is_liquid handles
  - whether you can eat it with a fork
  - whether it uses eatfood.ogg or drink.ogg
  - and all the messages pertaining to consuming such things, i.e., from [user] "eats" to [user] "drinks"

Also, eating stuff with utensils now properly spawns the trash item.

### CODE REVIEW

**Comment by alsoandanswer** [comment]
!bugfix

**Comment by alsoandanswer** [comment]
!review

File: code/game/objects/items/weapons/material/kitchen.dm [hunk 1]
```diff
@@ -53,7 +54,11 @@
 				return
 			M.visible_message("<span class='notice'>\The [user] feeds some [loaded] to \the [M] with \the [src].</span>")
 		reagents.trans_to_mob(M, reagents.total_volume, CHEM_INGEST)
-		playsound(M.loc,'sound/items/eatfood.ogg', rand(10,40), 1)
+		if(loadedis_liquid)
+			playsound(user.loc, 'sound/items/drink.ogg', rand(10, 50), 1)
```
Comment (mikomyazaki) [inline code/game/objects/items/weapons/material/kitchen.dm]:
```suggestion
			playsound(user.loc, 'sound/items/drink.ogg', rand(10, 50), 1)
			loadedis_liquid = FALSE
```

Maybe makes more sense for line 61 to be here?

File: code/modules/reagents/reagent_containers/food.dm [hunk 2]
```diff
@@ -7,9 +7,14 @@
 	volume = 50 //Sets the default container amount for all food items.
 	var/filling_color = "#FFFFFF" //Used by sandwiches
 	drop_sound = 'sound/items/drop/food.ogg'
+	var/is_liquid = FALSE // Handles whether you can eat with a fork & which eating sound is played
 
 /obj/item/reagent_containers/food/self_feed_message(var/mob/user)
 	to_chat(user, "<span class='notice'>You eat \the [src].</span>")
```
Comment (mikomyazaki) [inline code/modules/reagents/reagent_containers/food.dm]:
```suggestion
	to_chat(user, "<span class='notice'>You [is_liquid? "drink" : "eat"] \the [src].</span>")
	
```

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 3]
```diff
@@ -42,7 +42,7 @@
 
 /obj/item/reagent_containers/food/snacks/proc/on_consume(var/mob/eater, var/mob/feeder = null)
 	if(!reagents.total_volume)
-		eater.visible_message("<span class='notice'>[eater] finishes eating \the [src].</span>","<span class='notice'>You finish eating \the [src].</span>")
+		eater.visible_message("<span class='notice'>[eater] finishes [is_liquid ? "drinking" : "eating"] \the [src].</span>","<span class='notice'>You finish [is_liquid ? "drinking" : "eating"] \the [src].</span>")
```
Comment (mikomyazaki) [inline code/modules/reagents/reagent_containers/food/snacks.dm]:
```suggestion
		eater.visible_message(span("notice", "[eater] finishes [is_liquid ? "drinking" : "eating"] \the [src]."),span("notice","You finish [is_liquid ? "drinking" : "eating"] \the [src].")
```

Also in other places.

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 4]
```diff
@@ -202,31 +203,42 @@
...
+				if (U.reagents.total_volume > 0)
+					to_chat(user, "<span class='warning'>You already have \the [src] on \the [U].</span>")
+					return
+
+				user.visible_message( \
+					"\The [user] scoops up some of \the [src] with \the [U]!", \
+					"<span class='notice'>You scoop up some of \the [src] with \the [U]!</span>" \
+				)
+
+				src.bitecount++
```
Comment (mikomyazaki) [inline code/modules/reagents/reagent_containers/food/snacks.dm]:
```suggestion
				bitecount++
```

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 5]
```diff
@@ -202,31 +203,42 @@
...
+				U.cut_overlays()
+				U.loaded = "[src]"
+				var/image/I = new(U.icon, "loadedfood")
+				I.color = src.filling_color
+				U.add_overlay(I)
+
+				reagents.trans_to_obj(U, min(reagents.total_volume,5))
+				if(src.is_liquid)
```
Comment (mikomyazaki) [inline code/modules/reagents/reagent_containers/food/snacks.dm]:
```suggestion
				if(is_liquid)
```

**Comment by alsoandanswer** [comment]
> Did you check for other usages of the renamed items?

Yes, the recipes and mapping are all in order.

**Pull request merged**